src/transactions/TransactionFrame.cpp

```diff
@@ -740,6 +740,19 @@ TransactionFrame::applyOperations(SignatureChecker& signatureChecker,
                                   TransactionMeta& outerMeta)
 {
     ZoneScoped;
+    auto& internalErrorCounter = app.getMetrics().NewCounter(
+        {"ledger", "transaction", "internal-error"});
+
+    // We do not want to treat exception as internal-error for non-live ledger
```

Comment:
It looks like the counter is incremented and an error is logged for both `LEDGER_PROTOCOL_VERSION` and `LEDGER_PROTOCOL_VERSION - 1` ledger versions. I would mention that it's still possible to get an internal-error for a non-live ledger due to this.

I wonder if there's a better solution here though. If you have the `LedgerHeader` for the ledger you're catching up to, you could use that version, but I'm not sure if that's easy/worth it to get and expose here.